subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -971,16 +961,14 @@ public FileTree getCandidateClassFiles() {
         return getTestClassesDirs().getAsFileTree().matching(patternSet);
     }
 
+
     /**
-     * Allows filtering tests for execution.
-     *
-     * @return filter object
-     * @since 1.10
+     * {@inheritDoc}
      */
     @Incubating
-    @Nested
+    @Internal
     public TestFilter getFilter() {
-        return filter;
+        return super.getFilter();
```

Comment:
We should get rid of this override, and have a single, consistent API in `AbstractTest`